src/uu/join/src/join.rs

```diff
@@ -677,12 +677,29 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
     settings.line_ending = LineEnding::from_zero_flag(matches.get_flag("z"));
 
-    let file1 = matches.get_one::<String>("file1").unwrap();
-    let file2 = matches.get_one::<String>("file2").unwrap();
+    Ok(settings)
+}
 
+/// Validates the provided file inputs.
+///
+/// This function checks if both files are standard input and returns an error if they are.
+fn validate_files(file1: &str, file2: &str) -> UResult<()> {
     if file1 == "-" && file2 == "-" {
         return Err(USimpleError::new(1, "both files cannot be standard input"));
     }
+    Ok(())
+}
```

Comment:
I think it is overkill to have a function for this trivial functionality as it is no shared functionality.